compiler/next/include/chpl/uast/BracketLoop.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_BRACKETLOOP_H
+#define CHPL_UAST_BRACKETLOOP_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/IndexableLoop.h"
+#include "chpl/uast/WithClause.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a bracket loop. For example:
+
+  \rst
+  .. code-block:: chapel
+
+      // Example 1:
+      [i in 0..15] writeln(i);
+
+  \endrst
+
+ */
+class BracketLoop final : public IndexableLoop {
+ private:
+  BracketLoop(ASTList children, int8_t indexChildNum,
+         int8_t iterandChildNum,
+         int8_t withClauseChildNum,
+         int loopBodyChildNum,
+         int numLoopBodyStmts,
+         bool usesImplicitBlock,
+         bool isExpressionLevel)
```

Comment:
The indentation on these arguments is a little wonky.  Not a big deal, just pointing it out

Reply:
Fixed, thanks!